theseus/geometry/manifold.py

```diff
@@ -138,6 +140,13 @@ def __deepcopy__(self, memo):
         memo[id(self)] = the_copy
         return the_copy
 
+    # calls to() on the internal tensors
+    def to(self, *args, **kwargs):
+        _, dtype, *_ = torch._C._nn._parse_to(*args, **kwargs)
+        if dtype is not None:
+            _CHECK_DTYPE_SUPPORTED(dtype)
```

Comment:
I don't think we want this in `Variable` because auxiliary variables of other dtypes might be necessary. The current placement (in `Manifold` class) should cover all optimization variables.